utilities/rest-helper-factory.js

```diff
@@ -585,6 +590,285 @@ module.exports = function(logger, mongoose, server) {
         }
       })
     },
+    
+        /**
+     * Creates an endpoint for RECOVER /RESOURCE/{_id}/recover
+     * @param server: A Hapi server.
+     * @param model: A mongoose model.
+     * @param options: Options object.
+     * @param logger: A logging object.
+     */
+    generateRecoverOneEndpoint: function(server, model, options, logger) {
+      // This line must come first
+      validationHelper.validateModel(model, logger)
+      const Log = logger.bind(chalk.yellow('RecoverOne'))
+
+      const collectionName = model.collectionDisplayName || model.modelName
+      if (config.logRoutes) {
+        Log.note('Generating Recover One endpoint for ' + collectionName)
+      }
+
+      options = options || {}
+
+      let resourceAliasForRoute
+
+      if (model.routeOptions) {
+        resourceAliasForRoute = model.routeOptions.alias || model.modelName
+      } else {
+        resourceAliasForRoute = model.modelName
+      }
+
+      const handler = HandlerHelper.generateRecoverHandler(model, options, Log)
+
+      let payloadModel = null
+      /* if (config.enableSoftDelete) {
+        payloadModel = Joi.object({ hardDelete: Joi.bool() }).allow(null)
+
+        if (!config.enablePayloadValidation) {
+          payloadModel = Joi.alternatives().try(payloadModel, Joi.any())
+        }
+      } */
+
+      let auth = false
+      let recoverOneHeadersValidation = Object.assign(headersValidation, {})
+
+      if (config.authStrategy && model.routeOptions.recoverAuth !== false) {
+        auth = {
+          strategy: config.authStrategy
+        }
+
+        const scope = authHelper.generateScopeForEndpoint(model, 'recover', Log)
```

Comment:
This needs to be `const scope = authHelper.generateScopeForEndpoint(model, 'update', Log)` as we are treating this as an update action.